ibis/backends/base/sqlglot/compiler.py

```diff
@@ -1195,158 +1356,102 @@ def visit_JoinTable(self, op, *, parent, index):
-    ops.Repeat: "repeat",
-    ops.Reverse: "reverse",
-    ops.RowNumber: "row_number",
-    ops.Sign: "sign",
-    ops.Sin: "sin",
-    ops.Sqrt: "sqrt",
-    ops.StartsWith: "starts_with",
-    ops.StrRight: "right",
-    ops.StringAscii: "ascii",
-    ops.StringContains: "contains",
-    ops.StringLength: "length",
-    ops.StringReplace: "replace",
-    ops.StringSplit: "split",
-    ops.StringToTimestamp: "str_to_time",
-    ops.Tan: "tan",
-    ops.Translate: "translate",
-    ops.Unnest: "explode",
-    ops.Uppercase: "upper",
-}
-
-_BINARY_INFIX_OPS = {
-    # Binary operations
-    ops.Add: sge.Add,
-    ops.Subtract: sge.Sub,
-    ops.Multiply: sge.Mul,
-    ops.Divide: sge.Div,
-    ops.Modulus: sge.Mod,
-    ops.Power: sge.Pow,
-    # Comparisons
-    ops.GreaterEqual: sge.GTE,
-    ops.Greater: sge.GT,
-    ops.LessEqual: sge.LTE,
-    ops.Less: sge.LT,
-    ops.Equals: sge.EQ,
-    ops.NotEquals: sge.NEQ,
-    # Boolean comparisons
-    ops.And: sge.And,
-    ops.Or: sge.Or,
-    ops.Xor: sge.Xor,
-    # Bitwise business
-    ops.BitwiseLeftShift: sge.BitwiseLeftShift,
-    ops.BitwiseRightShift: sge.BitwiseRightShift,
-    ops.BitwiseAnd: sge.BitwiseAnd,
-    ops.BitwiseOr: sge.BitwiseOr,
-    ops.BitwiseXor: sge.BitwiseXor,
-    # Time arithmetic
-    ops.DateAdd: sge.Add,
-    ops.DateSub: sge.Sub,
-    ops.DateDiff: sge.Sub,
-    ops.TimestampAdd: sge.Add,
-    ops.TimestampSub: sge.Sub,
-    ops.TimestampDiff: sge.Sub,
-    # Interval Marginalia
-    ops.IntervalAdd: sge.Add,
-    ops.IntervalMultiply: sge.Mul,
-    ops.IntervalSubtract: sge.Sub,
-}
-
-for _op, _sym in _BINARY_INFIX_OPS.items():
-
-    def _fmt(self, op, *, _sym: sge.Expression = _sym, left, right):
-        return _sym(
-            this=self._add_parens(op.left, left),
-            expression=self._add_parens(op.right, right),
-        )
+    visit_DateSub = (
+        visit_DateDiff
+    ) = (
+        visit_TimestampSub
+    ) = visit_TimestampDiff = visit_IntervalSubtract = visit_Subtract
+
+    @parenthesize_inputs
+    def visit_Multiply(self, op, *, left, right):
+        return sge.Mul(this=left, expression=right)
+
+    visit_IntervalMultiply = visit_Multiply
+
+    @parenthesize_inputs
+    def visit_Divide(self, op, *, left, right):
+        return sge.Div(this=left, expression=right)
+
+    @parenthesize_inputs
+    def visit_Modulus(self, op, *, left, right):
+        return sge.Mod(this=left, expression=right)
+
+    @parenthesize_inputs
+    def visit_Power(self, op, *, left, right):
+        return sge.Pow(this=left, expression=right)
+
+    @parenthesize_inputs
+    def visit_GreaterEqual(self, op, *, left, right):
+        return sge.GTE(this=left, expression=right)
+
+    @parenthesize_inputs
+    def visit_Greater(self, op, *, left, right):
+        return sge.GT(this=left, expression=right)
 
-    setattr(SQLGlotCompiler, f"visit_{_op.__name__}", _fmt)
+    @parenthesize_inputs
+    def visit_LessEqual(self, op, *, left, right):
```

Comment:
Why are these spelled out instead of being generated like `SIMPLE_OPS`?  `BINARY_INFIX_OPS` could be a mapping.

Reply:
Because implementations call the base class impl. I don't want to keep refactoring this. Is this a blocker?

Reply:
Nope, it is not a blocker.